src/main/scala/com/exini/dicom/streams/DicomFlow.scala

```diff
@@ -355,7 +355,12 @@ trait TagPathTracking[Out]
     }
 
     abstract override def onItem(part: ItemPart): List[Out] = {
-      if (!inFragments) tagPath = tagPath.previous.thenItem(tagPath.tag, part.index)
+      if (!inFragments)
+        tagPath = tagPath match {
+          case t: TagPathItemEnd =>
+            t.previous.thenItem(t.tag, t.item + 1)
+          case t => t.previous.thenItem(t.tag, 1)
+        }
```

Comment:
This was the only place where item indices were used. It can be replaced with standard tag path logic.